pkg/kubelet/kubelet.go

```diff
@@ -2888,6 +2888,15 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod, podStatus *kubecontaine
 			return nil, err
 		}
 		allocatedPod = pod
+
+		// Special case when the updated allocation matches the actual resources. This can occur
+		// when reverting a resize that hasn't been actuated, or when making an equivalent change
+		// (such as CPU requests below MinShares).
+		if allocatedResourcesMatchStatus(allocatedPod, podStatus) {
```

Comment:
This seems like an optimization that we can also do without. Any strong reason to keep it?

Reply:
This always gets corrected on the next sync loop, but that could be 1 minute later. That means that the resize InProgress status actually resolves more slowly when it's not making any changes than when it is (since we wouldn't set a WatchCondition in this case)

Reply:
Not a hard requirement though.

Reply:
Hmm...we could also set the watch condition as an alternative. Just wanted to think about options before adding more special cases. Let's make it clear in the comment that this is an optimization and dropping it will only result in slower convergence.

Reply:
Updated the comment.